### ISSUE TITLE
Issue: JsonPathMatcher - not working for rootless jsons?

### ISSUE THREAD

**Issue by FSatmar** [WireMock-Net/WireMock.Net#100]
Hi,
Could you pls help me with defining a JsonPathMatcher for a json of the type:
```
{
   id: 1,
   name: "AwesomeName"
}
```

I have tried "$.[?(@.id == 1)]" but it doesn't seem to work.

I tried it with the following test as well:
```
            // given
            var spec = Request.Create().UsingAnyVerb().WithBody(new JsonPathMatcher("$.[?(@.id == 1)]"));

            // when
            string bodyAsString = "{  \"id\": 1, \"name\": \"AwesomeName\"  }";
            byte[] body = Encoding.UTF8.GetBytes(bodyAsString);
            var request = new RequestMessage(new Uri("http://localhost/foo"), "PUT", ClientIp, body, bodyAsString, Encoding.UTF8);

            // then
            var requestMatchResult = new RequestMatchResult();
            Check.That(spec.GetMatchingScore(request, requestMatchResult)).IsNotEqualTo(1.0);
```

**Comment by StefH** [comment]
When testing at http://jsonpath.herokuapp.com/

This works...
![json](https://user-images.githubusercontent.com/249938/36897858-95f904ce-1e18-11e8-8bcb-e5a32ead5b24.png)

So actually, I don't know what's wrong?

Or maybe use `.WithJsonBody` instead of `.WithBody()`

**Comment by FSatmar** [comment]
I tried it like this:
```

        [Test]
        public void Request_WithBodyJsonPathMatcher_true()
        {
            // given
            var spec = Request.Create().UsingAnyVerb().WithBody(new JsonPathMatcher("$.[?(@.id == 1)]"));

            var requestData = new JsonObject()
            {
                Id = 1,
                Name = "AwesomeName"
            };

            var bodyData = new BodyData {BodyAsJson = requestData};

            // when
            var request = new RequestMessage(new Uri("http://localhost/foo"), "POST", "::1", bodyData);

            // then
            var requestMatchResult = new RequestMatchResult();
            spec.GetMatchingScore(request, requestMatchResult).Should().Be(1);
        }

        [DataContract]
        public sealed class JsonObject
        {
            [DataMember(Order = 1)]
            public int Id { get; set; }
            [DataMember(Order = 2)]
            public string Name { get; set; }
        }
```
But still no luck.

**Comment by StefH** [comment]
For some reason:

It should be `"$..[?(@.Id == 1)]"`.
See linked unit-test.

Maybe a bit related to https://github.com/JamesNK/Newtonsoft.Json/issues/1256 ?

**Comment by StefH** [comment]
@FSatmar This is now fixed, please check new NuGet 1.0.3.12

**Comment by FSatmar** [comment]
Great, it's working with the double dots, thank you very much!
